src/data_server/s3service.rs

```diff
@@ -706,22 +713,50 @@ impl S3 for S3ServiceServer {
                 access_key: credentials.access_key,
             })
             .await
-            .unwrap()
-            .into_inner();
-        match user_client
-            .collection_service
+        {
+            Ok(collection) => collection.into_inner(),
+            Err(err) => {
+                log::error!("{}", err);
+                return Err(s3_error!(NoSuchBucket, "Bucket not found"));
+            }
+        };
+        let channel = match user_client.endpoint.connect().await {
+            Ok(channel) => channel,
+            Err(err) => {
+                log::error!("{}", err);
+                return Err(s3_error!(NotSignedUp, "Your account is not signed up"));
+            }
+        };
+        let mut client = collection_service_client::CollectionServiceClient::with_interceptor(
+            channel,
+            user_client.interceptor,
+        );
+
+        match client
             .update_collection(UpdateCollectionRequest {
                 collection_id: collection.collection_id,
                 labels: vec![KeyValue {
                     key: "apps.aruna-storage.org/cors".to_string(),
-                    value: serde_json::to_string(&cors).unwrap(),
+                    value: match serde_json::to_string(&cors) {
+                        Ok(value) => value,
+                        Err(err) => {
+                            log::error!("{}", err);
+                            return Err(s3_error!(
+                                InvalidArgument,
+                                "Invalid CORS headers provided"
+                            ));
+                        }
+                    },
                 }],
                 ..Default::default()
             })
             .await
         {
-            Ok(response) => Ok(S3Response::new(PutBucketCorsOutput {})),
-            Err(err) => Err(s3_error!(NotSignedUp, "Your Account is not signed up")),
+            Ok(_response) => Ok(S3Response::new(PutBucketCorsOutput {})),
+            Err(err) => {
+                log::error!("{}", err);
+                Err(s3_error!(NotSignedUp, "Your Account is not signed up"))
```

Comment:
Again, this is not the correct error. Additionally, you should be able to map your error to a `s3_error` and then return / bubble up it via `?` this might be more readable.

e.g.:

`.map_err(|_| s3_error!(NotSignedUp, "Unable to authenticate user"))?`